amd_openvx/openvx/ago/ago_haf_cpu_generic_functions.cpp

```diff
@@ -0,0 +1,878 @@
+                    *(vx_uint8 *)tmp_datap = (vx_uint8)filling_data;
+                }
+                else
+                {
+                    if (filling_data > INT16_MAX)
+                        filling_data = INT16_MAX;
+                    else if (filling_data < INT16_MIN)
+                        filling_data = INT16_MIN;
+                    *(vx_int16 *)tmp_datap = (vx_int16)filling_data;
+                }
+            }
+        }
+    }
+
+    status |= vxUnmapImagePatch(tmp, tmp_map_id);
+    status |= vxUnmapImagePatch(filling, filling_map_id);
+
+    status |=replicateConvolve(tmp, conv, upsample, border);
+
+    vx_rectangle_t upsample_rect;
+    vx_imagepatch_addressing_t upsample_addr = VX_IMAGEPATCH_ADDR_INIT;
+    vx_map_id upsample_map_id;
+    void * upsample_base = NULL;
+    vx_df_image upsample_format;
+
+    status |= vxQueryImage(upsample, VX_IMAGE_FORMAT, &upsample_format, sizeof(upsample_format));
+    status = vxGetValidRegionImage(upsample, &upsample_rect);
+    status |= vxMapImagePatch(upsample, &upsample_rect, 0, &upsample_map_id, &upsample_addr, (void **)&upsample_base, VX_READ_AND_WRITE, VX_MEMORY_TYPE_HOST, 0);
+
+    for (int ix = 0; ix < (int)width; ix++)
+    {
+        for (int iy = 0; iy < (int)height; iy++)
+        {
+            void* upsample_p = vxFormatImagePatchAddress2d(upsample_base, ix, iy, &upsample_addr);
+            vx_int32 upsample_data = upsample_format == VX_DF_IMAGE_U8 ? *(vx_uint8 *)upsample_p : *(vx_int16 *)upsample_p;
+            upsample_data *= 4;
+            if (upsample_format == VX_DF_IMAGE_U8)
+            {
+                if (upsample_data > UINT8_MAX)
+                    upsample_data = UINT8_MAX;
+                else if (upsample_data < 0)
+                    upsample_data = 0;
+                *(vx_uint8 *)upsample_p = (vx_uint8)upsample_data;
+            }
+            else
+            {
+                if (upsample_data > INT16_MAX)
+                    upsample_data = INT16_MAX;
+                else if (upsample_data < INT16_MIN)
+                    upsample_data = INT16_MIN;
+                *(vx_int16 *)upsample_p = (vx_int16)upsample_data;
+            }
+        }
+    }
+    status |= vxUnmapImagePatch(upsample, upsample_map_id);
+    status |= vxReleaseImage(&tmp);
+    return status;
+}
+
+int HafCpu_LaplacianPyramid_DATA_DATA_DATA
+    (
+        vx_node node,
+        vx_image input, 
+        vx_pyramid laplacian,
+        vx_image output
+    )
+{
+    vx_status status = VX_SUCCESS;
+    
+    vx_context context = vxGetContext((vx_reference)node);
+    
+    vx_size lev;
+    vx_size levels = 1;
+    vx_uint32 width = 0;
+    vx_uint32 height = 0;
+    vx_uint32 level_width = 0;
+    vx_uint32 level_height = 0;
+    vx_df_image format;
+    vx_enum policy = VX_CONVERT_POLICY_SATURATE;
+    vx_border_t border;
+    vx_convolution conv = 0;
+    vx_image pyr_gauss_curr_level_filtered = 0;
+    vx_image pyr_laplacian_curr_level = 0;
```

Comment:
Codacy found an issue: [Variable 'pyr_laplacian_curr_level' is assigned a value that is never used.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=6287409)